src/tools/bit2core.c

```diff
@@ -10,37 +10,42 @@
 
 const int CORE_HEADER_SIZE = 4096;
 const int BITSTREAM_HEADER_SIZE = 120;
+const int BITSTREAM_HEADER_TARGETNAME_LOC = 0x4C;
 
 unsigned char bitstream[MAX_MB*BYTES_IN_MEGABYTE];
 
 typedef struct {
   char name[8];
   int num;
   int max_core_mb_size;
+  char target_device_name[13];
 } rev_info;
 
+static rev_info revs[] =
+{
+  // name   num MB  Target Device Name
+  { "nexys", 1, 4, "7a100tcsg324" },
+  { "r2",    2, 4, "7a100tfgg484" },
+  { "r3",    3, 8, "7a200tfbg484" }
+};
+// NOTE: Append future revision details to end of this array
+
+
 void show_help(void)
 {
   fprintf(stderr, "MEGA65 bitstream to core file converter v0.0.2.\n"
                   "---------------------------------------\n"
```

Comment:
It would be better to auto-detect the target, where possible.
Also, don't forget megaphoner1 and megaphoner4 targets, both of which use a 7a100 part.
Might make sense to just put <target> <foo.bit> <core name> here, and then list the targets below, as you have done.  Also note that there are nexys are nexys4ddr variants.  

this reminds me, we should actually indicate the intended target for a .COR file in the 4KB header somewhere.  Then we can modify the flash menu to check that, before trying to flash it.

Reply:
> It would be better to auto-detect the target, where possible.

I'm finding this term 'target' a little overloaded. In some contexts, it seems to refer to the fpga-type, in other cases, it seems to refer to the mega65-device-type. Would be good to clarify which one you mean here, and also find some explicit terms to distinguish between these two. Perhaps "mega65-target" and "fpga-target"? I'm open to alternatives :)

So here, I'm guessing you mean, instead of my inclusion of the new "<r2|r3>" argument at the front (for the user to specify the mega65-target), you would rather not include such an argument and auto-detect the fpga-target instead.

On that point, my perspective is that it is better to do both, in an effort to prevent the user from shooting themselves in the foot, and that's what I've done thus far.

For example, the sort of case I'm trying to address with this:
- a novice user accidentally downloads a **nexys4ddr.bit** file (even though they have a devkit/r3)
- they run bit2core in its present state: `bit2core nexys4ddr.bit text1 text2 output.cor`
- then bit2core happily outputs a .cor file, and the novice user will blindly apply it onto their device, with no warning/error to caution him or stop him.

So my intent with that new 1st argument <nexys|r1|r2|etc> is to force the user to explicitly tell the tool what mega65-target they want this .cor file for. The tool can then assess the fpga-target within the bitstream header, and then warn the user, "no, this bitstream is not suitable for your mega65-target"

> this reminds me, we should actually indicate the intended target for a .COR file in the 4KB header somewhere. Then we can modify the flash menu to check that, before trying to flash it.

Agreed, so I've made a card #19  for that.

Reply:
Feel like the behaviour you describe is already in place, and I've made card #19 for your idea on the extra field in the .cor file's header. So I'll mark this thread as resolved.